torch/distributions/exp_family.py

```diff
@@ -58,3 +58,42 @@ def entropy(self):
         for np, g in zip(nparams, gradients):
             result -= np * g
         return result
+
+    @staticmethod
+    def _from_natural_params(*params):
+        raise NotImplementedError
+
+    def normalized_product(self, dim=-1, keepdim=False):
+        r"""
+        Returns a new distribution object whose probability mass/density function is the normalized product 
+        of the probability mass/density functions along given axis. The product of exponential family 
+        distributions results in a distribution in the same family, but with new natural parameters :math:`\sum(\theta)` 
+
+        .. math::
+
+            \frac{1}{c} \prod_i p_{F}(x; \theta_i) = \exp(\langle t(x), \sum_i\theta_i\rangle - F(\sum_i\theta_i) + k(x))
+
+        where c is the normalization constant and i is the index along given dimension.
+
+        For a batched distribution d and a single observation x::
+
+        d.prod(dim).log_prob(x) = d.log_prob(x).sum(dim) + c
```

Comment:
Is this rendered verbatim (i.e., as code as opposed to text)?

Reply:
It's supposed to be, yes. I don't know how to check rendering in local builds though. I indented it now, btw.